pkg/reconciler/route/route.go

```diff
@@ -569,3 +587,28 @@ func setTargetsScheme(rs *v1alpha1.RouteStatus, dnsNames []string, scheme string
 		}
 	}
 }
+
+func findMatchingCert(domains []string, certs []*netv1alpha1.Certificate) *netv1alpha1.Certificate {
+	for _, cert := range certs {
+		if certMatches(domains, cert) {
+			return cert
+		}
+	}
+	return nil
+}
+
+func certMatches(domains []string, cert *netv1alpha1.Certificate) bool {
+	dnsNames := sets.NewString()
+
+	for _, dns := range cert.Spec.DNSNames {
+		dnsNames.Insert(strings.SplitAfterN(dns, ".", 2)[1])
```

Comment:
Is this assuming that `dns` is a wildcard cert?  Should we call this method `wildcardCertMatches`?

Reply:
Yes it assumes a wildcard certificate. I'll rename these methods.